core/sdk-core/src/main/java/software/amazon/awssdk/core/internal/async/SplittingTransformer.java

```diff
@@ -237,6 +259,13 @@ public CompletableFuture<ResponseT> prepare() {
                     CompletableFutureUtils.forwardResultTo(upstreamFuture, resultFuture);
                 }
             }
+            resultFuture.whenComplete((r, e) -> {
+                if (e == null) {
+                    return;
+                }
+                isCancelled.set(true);
```

Comment:
Should we handle cancellation logic on line 125 here?

Reply:
actually, setting `isCancelled` to true is redundant with the `whenComplete` in the constructor, we can just propagate the error to `individualFuture` here and let the `whenComplete` in the constructor handle cancellation logic. Will remove this `isCancelled.set(true);`